rca/static_src/javascript/components/form-focus.js

```diff
@@ -13,51 +13,79 @@ class FormFocus {
         this.bindEvents();
     }
 
-    // Apply focus class
-    applyClass() {
-        this.formItem.classList.add(this.hasFocusClass);
-    }
-
-    // Remove focus class
-    removeClass() {
-        // Check if input has content and add content class if it does
+    // Apply state class
+    applyStateClass() {
         if (this.formType === 'textarea') {
-            if (this.formFieldTextarea.value) {
+            if (
+                this.formFieldTextarea.value &&
+                this.formFieldTextarea === document.activeElement
+            ) {
+                this.formItem.classList.add(
+                    this.hasContentClass,
+                    this.hasFocusClass,
+                );
+            } else if (this.formFieldTextarea.value) {
                 this.formItem.classList.add(this.hasContentClass);
                 this.formItem.classList.remove(this.hasFocusClass);
+            } else if (this.formFieldTextarea === document.activeElement) {
+                this.formItem.classList.add(this.hasFocusClass);
             } else {
-                this.formItem.classList.remove(this.hasFocusClass);
-                this.formItem.classList.remove(this.hasContentClass);
+                this.formItem.classList.remove(
+                    this.hasFocusClass,
+                    this.hasContentClass,
+                );
             }
         }
 
         if (this.formType === 'input') {
-            if (this.formFieldInput.value) {
+            if (
+                this.formFieldInput.value &&
+                this.formFieldInput === document.activeElement
+            ) {
+                this.formItem.classList.add(
+                    this.hasContentClass,
+                    this.hasFocusClass,
+                );
+            } else if (this.formFieldInput.value) {
                 this.formItem.classList.add(this.hasContentClass);
                 this.formItem.classList.remove(this.hasFocusClass);
+            } else if (this.formFieldInput === document.activeElement) {
+                this.formItem.classList.add(this.hasFocusClass);
             } else {
-                this.formItem.classList.remove(this.hasFocusClass);
-                this.formItem.classList.remove(this.hasContentClass);
+                this.formItem.classList.remove(
+                    this.hasFocusClass,
+                    this.hasContentClass,
+                );
```

Comment:
the two functions in `applyStateClass` look pretty similar. you could do something like the following for less repetition

```
applyStateClass() {
    if (this.formType === 'textarea') {
        this.doSomething(this.formFieldTextarea);
    }

    if (this.formType === 'input') {
        this.doSomething(formFieldInput);
    }
}

doSomething(el) {
    if (
        el.value &&
        el === document.activeElement
    ) {
        this.formItem.classList.add(
            this.hasContentClass,
            this.hasFocusClass,
        );
    } else if (el.value) {
        this.formItem.classList.add(this.hasContentClass);
        this.formItem.classList.remove(this.hasFocusClass);
    } else if (el === document.activeElement) {
        this.formItem.classList.add(this.hasFocusClass);
    } else {
        this.formItem.classList.remove(
            this.hasFocusClass,
            this.hasContentClass,
        );
    }
}
```

on the other hand though you may want to keep them separate so it's easier to read and more customisable in the future : )

Reply:
Thanks man!